src/client/componentsMealsharing/AddMeal.js

```diff
@@ -0,0 +1,142 @@
+import "./Meal.css";
+import React, { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+
+export const AddMeal = () => {
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState("");
+  const [maxreservation, setMaxreservation] = useState("");
+  const [price, setPrice] = useState("");
+  const [created, setCreated] = useState([]);
+  const [meals, setMeals] = useState([]);
+  const [succeed, setSucceed] = useState("");
+
+  useEffect(() => {
+    fetchDataResult();
+  }, []);
+
+  const fetchDataResult = async () => {
+    const response = await fetch("http://localhost:3000/api/meals");
+    const Data = await response.json();
+    setMeals(Data);
+  };
+
+  const onhandleReservation = async (e) => {
+    const addId = meals.length + 1;
```

Comment:
I would add the id on the backend side and not let the client submit the id, as the database can automatically increment the id

Reply:
here I did not let the client submit the id, I put it inside the function, The client just needs to submit   Meal Title, Max seats, Meal Price, Meal Date, and Description. .)